### ISSUE TITLE
New Vardiff Order

### ISSUE THREAD

**Issue by litecoinmessiah** [moopless/stratum-mining-litecoin#40]
Can we have a new vardiff order so we don't have random share diffs but could set intervals from the base share diff, such as

VDIFF_BASE_TARGET = 16 # Base Share Difficulty
VDIFF_INTERVALS = 4 # Number of times the base share can double, i.e; 4 = 16, 32, 64, 128

then divided the gap between VDIFF_TARGET_TIME and VDIFF_RETARGET_TIME by VDIFF_INTERVALS.

to get the VDIFF_VARIANCE_PERCENT which would trigger each diff

Might be better for some pool operators to integrate vardiff this way, as the variance would be predictable.

Thank you for all your work, Much appreciated :)

**Comment by moopless** [comment]
I like this idea. Right now we have a bloolean for VARDIFF, it enables or disables it. What I would like to see is multiple implementations of VARDIFF that can be instantiated. This gives the pool operator more flexibility on the pool. 

I will get this implemented in the upcoming week or so.

**Comment by litecoinmessiah** [comment]
yeah manual/auto column added next to diff would be nice too :) thanks

**Comment by ahmedbodi** [comment]
added support in https://github.com/ahmedbodi/stratum-mining
